projects/gloo/api/v1/settings.proto

```
@@ -688,4 +691,39 @@ message ConsoleOptions {
     // Whether to enable the GraphQL API Explorer. This feature requires a Gloo Edge Enterprise license with GraphQL enabled.
     // Defaults to true.
     google.protobuf.BoolValue api_explorer_enabled = 2;
+}
+
+// GraphQL settings used by the control plane and UI.
+message GraphqlOptions {
+
+    // Options for how to validate changes to schema definitions.
+    SchemaChangeValidationOptions schema_change_validation_options = 1;
+
+    message SchemaChangeValidationOptions {
+        // Schema definition updates can be considered safe, dangerous, or breaking.
+        // If this field is set to true, then breaking schema updates will be rejected.
+        // Defaults to false.
+        google.protobuf.BoolValue reject_breaking_changes = 1;
+
+        // We use [GraphQL Inspector](https://www.graphql-inspector.com/docs/essentials/diff) to detect breaking changes
+        // to GraphQL schemas. This field allows for passing [processing rules](https://www.graphql-inspector.com/docs/essentials/diff#rules)
+        // to GraphQL Inspector to customize how various change types are handled.
+        repeated ProcessingRule processing_rules = 2;
+
+        enum ProcessingRule {
+            RULE_UNSPECIFIED = 0;
+
+            // Turn every dangerous change into a breaking change.
+            RULE_DANGEROUS_BREAKING = 1;
+
+            // Treat the removal of a deprecated field as a dangerous change, instead of a breaking change.
+            RULE_SUPPRESS_REMOVAL_OF_DEPRECATED_FIELD = 2;
```

Nit: Could this be renamed to something like `RULE_BREAK_ON_REMOVE_DEPRECATED_FIELD`? Suppress could go either way (suppressing the errors on removing deprecated fields, or suppressing changes when removing deprecated fields)

---

hmm, how about something like RULE_REMOVAL_OF_DEPRECATED_FIELD_DANGEROUS?

---

I think that would be good!